tools/Dashboard/DevHome.Dashboard/Views/DashboardView.xaml.cs

```diff
@@ -52,6 +52,7 @@ public partial class DashboardView : ToolPage, IDisposable
     private static DispatcherQueue _dispatcherQueue;
     private readonly ILocalSettingsService _localSettingsService;
     private readonly IWidgetExtensionService _widgetExtensionService;
+    private CancellationTokenSource _initWidgetsCancellationTokenSource;

```

Comment:
I like this idea in general. Having a token for unload/navigating away from a page is useful in dashboard and other pages too which require some compute heavy processing to fully populate.